src/App.js

```diff
@@ -122,96 +207,29 @@ export default class App extends React.Component {
+            stopMetronome={() => this.stopMetronome()}
           />
         </div>
-        <header className="App__header">
-          <div className="App__button-row">
-            <Button variant="contained" color="primary" onClick={() => {
-              if (numRows) {
-                this.metronome.current.play();
-              } else {
-                this.generateRow(4 - numRows);
-              }
-            }}>
-              {numRows ? (this.state.playing ? 'Stop' : 'Play') : 'Generate Notes'}
-            </Button>
-            <Button variant="contained" color="primary" onClick={async () => {
-              if (this.metronome.current.isPlaying()) {
-                await this.metronome.current.play();
-              }
-              this.setState({
-                highlightNote: -1,
-                highlightRow: 0,
-              });
-              this.metronome.current.play();
-            }}>
-              {numRows ? 'Start Over' : 'Generate and Play'}
-            </Button>
-            <Button variant="contained" color="primary" onClick={async () => {
-              if (this.metronome.current.isPlaying()) {
-                await this.metronome.current.play();
-              }
-              this.resetRows();
-            }}>
-              Clear
-            </Button>
-          </div>
-          <div className="App__note-display--container">
-            {this.state.rows.map((row, indexRow) => (
-              <div ref={!indexRow && this.firstBar} key={row.id} className="App__note-display--row">
-                <span className="App__note-display--clef">&#x1d122;</span>
-                {row.data.map((note, indexNote) => {
-                  const isCurrent = this.state.highlightRow === indexRow && this.state.highlightNote === indexNote;
-                  const isNext = nextHighlightedRow === indexRow && nextHighlightedNote === indexNote;
-                  return (
-                    <div key={indexNote} className="App__note-display--note-container">
-                      <div className={`App__note-display--note ${isCurrent && 'highlight'} ${isNext && 'next-highlight'}`}>
-                        {note}
-                      </div>
-                    </div>
-                  );
-                })}
-              </div>
-            ))}
-          </div>
-          <Snackbar open={this.state.errorShown} onClose={() => this.setState({ errorShown: false })}>
-            <Alert onClose={() => this.setState({ errorShown: false })} severity="error">
-              {this.state.error}
-            </Alert>
-          </Snackbar>
-        </header>
+        <div className="App__header">
+          <MusicSheet
+            play={() => this.play(numRows)}
+            playing={this.state.playing}
+            numRows={numRows}
+            startOver={() => this.startOver()}
+            clearRows={() => this.clearRows()}
+            renderRows={() => this.renderRows(nextHighlightedRow, nextHighlightedNote)}
```

Comment:
Maybe it makes sense the rendering of these to be handled inside the MusicSheet? Only pass down the highlighted notes

Reply:
The whole rendering logic needs to be moved to the component. But because it will also need to be reworked so we can reuse it for the other exercises, I've left that out to when we get to the rework.